### PR TITLE
Add UT for context Injection for predicate and projection

### LINKED ISSUE #16957: Predicates do not have managed context injection when predicate is local/not serialized
a placeholder for coming PR.

When a Predicate is created that expects to have Dep Injection with ManagedContext, if the Predicate is only used in the local JVM, and not serialized to another node, then the Predicate does not receive Dep Injection with managed context.

cc @mmedenjak

### PR DESCRIPTION
**Pull request by aditya-32**
Fixes #16957 
Added Unit test and code changes for testing context Injection for Predicates and projection in IMap 

Checklist:
- [ ] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [ ] Label `Add to Release Notes` or `Not Release Notes content` set
- [ ] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc

### CODE REVIEW

**Comment by aditya-32** [comment]
hi @ahmetmircik  can you please review this PR

**Comment by srknzl** [comment]
Added @ahmetmircik as review as per the request of the pr owner.

File: hazelcast/src/test/java/com/hazelcast/projection/MapProjectionTest.java [hunk 1]
```diff
@@ -120,6 +124,15 @@ public void projection_3Nodes_nullReturningProjection() {
         assertThat(result, containsInAnyOrder((Double) null, null, null));
     }
 
+    @Test
+    public void testProjectionContextInjection() {
+        IMap<String, Double> map = getMapWithNodeCount(1);
+        populateMap(map);
+        Collection<Double> result = map.project(new ProjectionWithContext());
+        assertTrue("Projection must be injected with ManagedContext ", isProjectionContextInjected.get());
+    }
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/projection/MapProjectionTest.java:133]: You dont add new logic for `MapProxyImpl.project`. Why add a new test?

File: hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java [hunk 2]
```diff
@@ -713,7 +713,7 @@ private EventJournalInitialSubscriberState subscribeToEventJournal(
      *                      if the projection is {@code null} or it is the identity projection
      * @return the future with the filtered and projected journal items
      */
-    private <K, V, PROJ_TYPE> CompletionStage<ReadResultSet<PROJ_TYPE>> readFromEventJournal(
+    public <K, V, PROJ_TYPE> CompletionStage<ReadResultSet<PROJ_TYPE>> readFromEventJournal(
```
Comment (srknzl) [inline hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java:716]: Let's do `protected` instead.

**Comment by srknzl** [comment]
run-lab-run

**Comment by srknzl** [comment]
run-lab-run

**Review by srknzl — APPROVED** [review]

**Comment by aditya-32** [comment]
hi @srknzl  the pr builder shows it failed no failure log is present ?

**Review by ahmetmircik — APPROVED** [review]
LGTM

**Comment by srknzl** [comment]
run-lab-run

**Comment by srknzl** [comment]
In the test some container start problem occurred, restarted it.

**Comment by aditya-32** [comment]
Hi can we get this pr merged?
cc: @srknzl @ahmetmircik

**Pull request merged**